lib/util/get-typescript-extension-map.js

```diff
@@ -0,0 +1,81 @@
+"use strict"
+
+const DEFAULT_MAPPING = normalise([
+    ["", ".js"],
+    [".ts", ".js"],
+    [".cts", ".cjs"],
+    [".mts", ".mjs"],
+    [".tsx", ".jsx"],
+])
+
+/**
+ * @typedef {Object} ExtensionMap
+ * @property {Record<string, string>} forward Convert from typescript to javascript
+ * @property {Record<string, string[]>} backward Convert from javascript to typescript
+ */
+
+function normalise(typescriptExtensionMap) {
+    const forward = {}
+    const backward = {}
+    for (const [typescript, javascript] of typescriptExtensionMap) {
+        forward[typescript] = javascript
+        if (!typescript) {
+            continue
+        }
+        backward[javascript] ??= []
+        backward[javascript].push(typescript)
+    }
+    return { forward, backward }
+}
+
+/**
+ * Gets `typescriptExtensionMap` property from a given option object.
+ *
+ * @param {object|undefined} option - An option object to get.
+ * @returns {ExtensionMap} The `typescriptExtensionMap` value, or `null`.
+ */
+function get(option) {
+    if (
+        option &&
+        option.typescriptExtensionMap &&
+        Array.isArray(option.typescriptExtensionMap)
```

Comment:
nits: can be simply rewritten to `Array.isArray(option?.typescriptExtensionMap)`, as  the package has dropped node.js < v16.

Reply:
Oh, I am totally going to fix this in an other PR!

I just followed what was there :eyes: